perception/autoware_tensorrt_bevdet/src/bevdet_node.cpp

```diff
@@ -0,0 +1,342 @@
+  CHECK_CUDA(cudaMalloc(&temp_gpu, width * height * 3));
+
+  for (size_t i = 0; i < imgs.size(); i++) {
+    cv::cvtColor(imgs[i], imgs[i], cv::COLOR_BGR2RGB);
+    CHECK_CUDA(cudaMemcpy(temp_gpu, imgs[i].data, width * height * 3, cudaMemcpyHostToDevice));
+    convert_RGBHWC_to_BGRCHW(temp_gpu, out_imgs + i * width * height * 3, 3, height, width);
+  }
+  delete[] temp;
+  CHECK_CUDA(cudaFree(temp_gpu));
+}
+
+void TRTBEVDetNode::callback(
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_f_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_fr_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_bl_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_b_img,
+  const sensor_msgs::msg::Image::ConstSharedPtr & msg_br_img)
+{
+  cv::Mat img_fl, img_f, img_fr, img_bl, img_b, img_br;
+  std::vector<cv::Mat> imgs;
+  img_fl = cv_bridge::toCvShare(msg_fl_img, "bgr8")->image;
+  img_f = cv_bridge::toCvShare(msg_f_img, "bgr8")->image;
+  img_fr = cv_bridge::toCvShare(msg_fr_img, "bgr8")->image;
+  img_bl = cv_bridge::toCvShare(msg_bl_img, "bgr8")->image;
+  img_b = cv_bridge::toCvShare(msg_b_img, "bgr8")->image;
+  img_br = cv_bridge::toCvShare(msg_br_img, "bgr8")->image;
+
+  imgs.emplace_back(img_fl);
+  imgs.emplace_back(img_f);
+  imgs.emplace_back(img_fr);
+  imgs.emplace_back(img_bl);
+  imgs.emplace_back(img_b);
+  imgs.emplace_back(img_br);
+
+  image_transport(imgs, imgs_dev_, img_w_, img_h_);
+
+  // uchar *imgs_dev
+  sampleData_.imgs_dev = imgs_dev_;
+
+  std::vector<Box> ego_boxes;
+  ego_boxes.clear();
+  float time = 0.f;
+
+  bevdet_->doInfer(sampleData_, ego_boxes, time);
+
+  autoware_perception_msgs::msg::DetectedObjects bevdet_objects;
+  bevdet_objects.header.frame_id = "base_link";
+  bevdet_objects.header.stamp = msg_f_img->header.stamp;
+
+  box3DToDetectedObjects(ego_boxes, bevdet_objects, class_names_, score_thre_);
+
+  pub_boxes_->publish(bevdet_objects);
+}
+
+void TRTBEVDetNode::camera_info_callback(int idx, const sensor_msgs::msg::CameraInfo::SharedPtr msg)
+{
+  if (caminfo_received_[idx])
+    return;  // already received;  not expected to modify because of we init the model only once
+
+  if (!initialized_) {  // get image width and height
+    img_w_ = msg->width;
+    img_h_ = msg->height;
+    initialized_ = true;
+  }
+  Eigen::Matrix3f intrinsics;
+  getCameraIntrinsics(msg, intrinsics);
+  cams_intrin[idx] = intrinsics;
+
+  Eigen::Quaternion<float> rot;
+  Eigen::Translation3f translation;
+  getTransform(
+    tf_buffer_->lookupTransform("base_link", msg->header.frame_id, rclcpp::Time(0)), rot,
+    translation);
+  cams2ego_rot[idx] = rot;
+  cams2ego_trans[idx] = translation;
+
+  caminfo_received_[idx] = true;
+  camera_info_received_flag_ =
+    std::all_of(caminfo_received_.begin(), caminfo_received_.end(), [](bool i) { return i; });
+}
+
+TRTBEVDetNode::~TRTBEVDetNode()
+{
+  delete imgs_dev_;
```

Comment:
The destructor `~TRTBEVDetNode()` uses `delete imgs_dev_;`.  However, `imgs_dev_` is allocated using `cudaMalloc`, which requires `cudaFree` for deallocation.  Replacing `delete imgs_dev_;` with `CHECK_CUDA(cudaFree(imgs_dev_));` ensures proper CUDA memory cleanup.  Failure to do so can lead to memory leaks and potential crashes.